tests/integration/openconfig-bgp/run.py

```diff
@@ -1,126 +1,71 @@
-  cmd += " --use-xpathhelper"
-  for i in ["openconfig-bgp"]:
-    cmd += " %s" % os.path.join(this_dir, "openconfig", "%s.yang" % i)
-  os.system(cmd)
-
-
-def teardown_test():
-  global this_dir
-  global del_dirs
-  del_dirs.append(os.path.join(this_dir, "ocbind"))
-  for dirname in del_dirs:
-    for root, dirs, files in os.walk(os.path.join(dirname), topdown=False):
-      for name in files:
-        os.remove(os.path.join(root, name))
-      for name in dirs:
-        os.rmdir(os.path.join(root, name))
-    os.rmdir(dirname)
-
-
-class PyangbindOCBGP(unittest.TestCase):
-  def __init__(self, *args, **kwargs):
-    unittest.TestCase.__init__(self, *args, **kwargs)
-
-    import ocbind
-    self.ph = YANGPathHelper()
-    self.instance = ocbind.openconfig_bgp(path_helper=self.ph)
+from pyangbind.lib.xpathhelper import YANGPathHelper
+from tests.base import PyangBindTestCase
+
+
+class OpenconfigBGPTests(PyangBindTestCase):
+  yang_files = [os.path.join('openconfig', 'openconfig-bgp.yang')]
+  pyang_flags = [
+    '-p %s' % os.path.join(os.path.dirname(__file__), "include"),
+    '--use-xpathhelper'
+  ]
+  split_class_dir = True
+  module_name = 'ocbind'
+
+  remote_yang_files = [
+    {
+      'local_path': 'include',
+      'remote_prefix': 'https://raw.githubusercontent.com/robshakir/yang/master/standard/ietf/RFC/',
+      'files': [
+        'ietf-inet-types.yang',
+        'ietf-yang-types.yang',
+        'ietf-interfaces.yang',
+      ]
+    },
+    {
+      'local_path': 'include',
+      'remote_prefix': 'https://raw.githubusercontent.com/openconfig/public/master/release/models/',
+      'files': [
+        'policy/openconfig-policy-types.yang',
+        'openconfig-extensions.yang',
+        'types/openconfig-types.yang',
+        'types/openconfig-inet-types.yang',
+        'types/openconfig-yang-types.yang',
+      ]
+    },
+    {
+      'local_path': 'openconfig',
+      'remote_prefix': 'https://raw.githubusercontent.com/openconfig/public/master/release/models/',
+      'files': [
+        'policy/openconfig-routing-policy.yang',
+        'bgp/openconfig-bgp-common-multiprotocol.yang',
+        'bgp/openconfig-bgp-common-structure.yang',
+        'bgp/openconfig-bgp-common.yang',
+        'bgp/openconfig-bgp-global.yang',
+        'bgp/openconfig-bgp-neighbor.yang',
+        'bgp/openconfig-bgp-peer-group.yang',
+        'bgp/openconfig-bgp-policy.yang',
+        'bgp/openconfig-bgp-types.yang',
+        'bgp/openconfig-bgp.yang',
+        'bgp/openconfig-bgp-errors.yang',
+        'interfaces/openconfig-interfaces.yang',
+      ]
+    }
+  ]
+
+  def setUp(self):
+    self.yang_helper = YANGPathHelper()
+    self.instance = self.ocbind.openconfig_bgp(path_helper=self.yang_helper)
 
   def test_001_add_bgp_neighbor(self):
     self.instance.bgp.neighbors.neighbor.add("192.0.2.1")
-    self.assertNotEqual(len(self.instance.bgp.neighbors.neighbor), 0)
+    self.assertGreater(len(self.instance.bgp.neighbors.neighbor), 0)
```

Comment:
Is `assertGreater` quite right here? This seems like it'd not catch a bug whereby the above line added two entries?